src/hotspot/cpu/s390/s390.ad

```diff
@@ -182,6 +185,169 @@ register %{
   reg_def Z_F15  (SOC, SOE, Op_RegF, 15, Z_F15->as_VMReg());
   reg_def Z_F15_H(SOC, SOE, Op_RegF, 99, Z_F15->as_VMReg()->next());
 
+// ----------------------------
+// Vector Registers
+// ----------------------------
+  // 1st 16 VRs are aliases for the FPRs which are already defined above.
+  reg_def Z_VR0   ( SOC, SOC, Op_RegF, 0, VMRegImpl::Bad());
```

Comment:
I think `Op_VecX` should be used. The _H, _J, _K versions shouldn't be needed.

Reply:
Thanks, Working on this.